suite-common/bluetooth/tests/bluetoothReducer.test.ts

```diff
@@ -0,0 +1,243 @@
+import { configureMockStore, extraDependenciesMock } from '@suite-common/test-utils';
+import { deviceActions } from '@suite-common/wallet-core';
+import { Device } from '@trezor/connect';
+
+import {
+    BluetoothDeviceState,
+    bluetoothAdapterEventAction,
+    bluetoothConnectDeviceEventAction,
+    bluetoothKnownDevicesUpdateAction,
+    bluetoothNearbyDevicesUpdateAction,
+    bluetoothRemoveKnownDeviceAction,
+    prepareBluetoothReducerCreator,
+} from '../src';
+import { BluetoothDeviceCommon, BluetoothState } from '../src/bluetoothReducer';
+
+const bluetoothReducer =
+    prepareBluetoothReducerCreator<BluetoothDeviceCommon>()(extraDependenciesMock);
+
+const initialState: BluetoothState<BluetoothDeviceCommon> = {
+    adapterStatus: 'unknown',
+    scanStatus: 'idle',
+    nearbyDevices: [] as BluetoothDeviceState<BluetoothDeviceCommon>[],
+    knownDevices: [] as BluetoothDeviceCommon[],
+};
+
+describe('bluetoothReducer', () => {
+    it('sets the bluetooth adapter as enabled/disabled when powered/unpowered', () => {
+        const store = configureMockStore({
+            extra: {},
+            reducer: combineReducers({ bluetooth: bluetoothReducer }),
+            preloadedState: { bluetooth: initialState },
+        });
+
+        expect(store.getState().bluetooth.adapterStatus).toEqual('unknown');
+        store.dispatch(bluetoothAdapterEventAction({ isPowered: true }));
+        expect(store.getState().bluetooth.adapterStatus).toEqual('enabled');
+        store.dispatch(bluetoothAdapterEventAction({ isPowered: false }));
+        expect(store.getState().bluetooth.adapterStatus).toEqual('disabled');
+    });
+
+    it('sorts the devices based on the `lastUpdatedTimestamp` and keeps the status for already existing device', () => {
+        const alreadyExistingNearbyDevice: BluetoothDeviceState<BluetoothDeviceCommon> = {
+            device: {
+                id: 'A',
+                data: [],
+                name: 'Trezor A',
+                lastUpdatedTimestamp: 1,
+            },
+            status: { type: 'pairing' },
+        };
+
+        const nearbyDeviceToBeDropped: BluetoothDeviceState<BluetoothDeviceCommon> = {
+            device: {
+                id: 'B',
+                data: [],
+                name: 'Trezor B',
+                lastUpdatedTimestamp: 2,
+            },
+            status: null,
+        };
+
+        const store = configureMockStore({
+            extra: {},
+            reducer: combineReducers({ bluetooth: bluetoothReducer }),
+            preloadedState: {
+                bluetooth: {
+                    ...initialState,
+                    nearbyDevices: [nearbyDeviceToBeDropped, alreadyExistingNearbyDevice],
+                },
+            },
+        });
+
+        const nearbyDevices: BluetoothDeviceCommon[] = [
+            {
+                id: 'A',
+                data: [],
+                name: 'Trezor A',
+                lastUpdatedTimestamp: 1,
+            },
```

Comment:
Nit to make shorter & make clear that this is still the same device.
Although now that I see the rest of the file, you could extract Trezor A as const at beginning of file. It's repeated often.
```suggestion
            alreadyExistingNearbyDevice.device,
```

Reply:
Food point https://github.com/trezor/trezor-suite/pull/17090/commits/6a30f1ff5805850c148192dc271585728f750d1a